loader.qml

```diff
@@ -136,47 +121,123 @@ ApplicationWindow {
                     text: qsTr("Screenshot")
                     ToolTip.visible: hovered
                     ToolTip.delay: 600
-                    ToolTip.text: qsTr("Take screenshot and store it named for use in /.thumbnail")
-                    onClicked: watchfaceDisplayFrame.snapshot()
+                    ToolTip.text: qsTr("Take a 640px screenshot and store it as PNG image")
+                    onClicked: roundCheckBox.checked ? watchfaceDisplayFrame.snapshot("-screenshot.png") : watchfaceDisplayFrame.snapshot(".png")
                 }
 
                 Button {
-                    id: reloadButton
+                    id: assetsButton
+
+                    property real sequencer: 0
+
+                    function transSnapshots() {
+                        if (sequencer === 0) {
```

Comment:
Uhuh.

Well, I’d accept this funky logic if this unblocks 2.0 but I feel there would be a much easier way to do all of this.
One proposal: an ultra simple QML file that only has:
- a Loader to show the watchface provided in Qt.application.arguments[1]
- an Image item to show a background provided in Qt.application.arguments[2]
- a Component.onCompleted() hook that calls grabToImage, saveToFile on Qt.application.arguments[3] and Qt.quit.
You’d invoke qmlscene on that with -platform offscreen such that no window even has to open.
Then, the wrapper shell script could call that in a loop to directly generate all previews without manual intervention and the same code could be later reused on the watches to do previews generation. The overall logic would also be a lot simpler to follow

Reply:
Ahah, you sure got me. I was obviously trying to force sequential thinking into a declarative language.
This simple sequencer and timer approach was the easiest way for me to circumvent the problem that grapToImage takes ~30ms to complete. I was very sure there must be a better way, embracing the qt way more.
Will look into your hints.

Reply:
Ok. Do you want to address this as part of this PR or separately later ?

Reply:
I do love to. But frankly could not give an ETA. Looking at the list of things i plan to do before 2.0 (Video, 3x3 launcher, triangular wallpapers, settings pages polishing, and some smaller things) I'd ask to do that with lower priority.
I'd create an issue on that rudimentary implementation of the screenshot function and assign myself so i do not forget. Ideally that could even be a dear collaboration between @beroset and me as soon as his head is out of cmake & bluez matters?